scylla-cql/src/errors.rs

```diff
@@ -474,6 +482,19 @@ pub enum BadKeyspaceName {
     IllegalCharacter(String, char),
 }
 
+/// An error that occurred when selecting a node connection
+/// to perform a request on.
+#[derive(Error, Debug, Clone)]
+#[non_exhaustive]
+pub enum ConnectionPoolError {
+    #[error("The pool is broken; Last connection failed with: {last_connection_error}")]
+    Broken { last_connection_error: String },
```

Comment:
I'm against stringifying. In any case when we don't want to leak a private error type to the user, we can use `Arc<dyn Error>`, which gives:
- cheap clones,
- we retain ability to downcast and this way test that a proper error arises.

Reply:
I was waiting for this discussion. I obviously wanted to make it `Arc<dyn Error>`, or even to expose a typed `ConnectionError` to the users.

However, when creating `ConnectionPoolError::Broken`, we only have access to `&ConnectionError`. To make it owned, we obviously need to clone it. So now the question is whether we want to stringify (current version), or derive `Clone` for `ConnectionError`, at the cost of wrapping underlying `std::io::Error`s with `Arc` (since `std::io::Error` is not `Clone`). WDYT?

Reply:
I'm OK with Arc'ing non-Clone errors. After all, those errors are not something on a hot-happy path.

Reply:
Changed `last_connection_error` to `Arc<dyn Error + Sync + Send>`.

Reply:
It got changed to `ConnectionError`.